Dropdown/Dropdown.js

```diff
@@ -269,147 +317,68 @@ const DropdownBase = kind({
+		delete rest.rtl;
+
 		const ariaProps = extractAriaProps(rest);
-		const dropdownButtonClassName = classnames(css.dropdownButton, {[css.upDropdownButton]: adjustedDirection === 'up'});
-		const opened = !disabled && open;
-		const transitionContainerClassName = classnames(css.transitionContainer, {[css.openTransitionContainer]: open, [css.upTransitionContainer]: adjustedDirection === 'up'});
-		const [DropDownButton, dropDownButtonProps, wrapperProps, skinVariants, groupProps, iconComponent] = (skin === 'silicon') ? [
-			Button,
-			{icon: open ? 'arrowlargeup' : 'arrowlargedown', iconPosition: 'after', minWidth: true},
-			{className: dropdownButtonClassName},
-			{'night': false},
-			{childComponent: RadioItem, itemProps: {size: 'small', className: css.dropDownListItem, css}, selectedProp: 'selected'},
-			[]
-		] : [
-			Item,
-			{},
-			{},
-			{},
-			{childComponent: Item},
-			[<Icon slot="slotAfter" key="icon" className={css.icon} size="small">{open ? 'arrowlargeup' : 'arrowlargedown'}</Icon>]
-
-		];
+		const calcAriaProps = ariaLabel != null ? null : {role: 'region', 'aria-labelledby': ariaLabelledBy};
+
+		const popupProps = {'aria-live': null, children, direction, disabled, onSelect, open, selected, skin, skinVariants: skin === 'silicon' ? {'night': false} : {}, width, role: null};
+
+		// `ui/Group`/`ui/Repeater` will throw an error if empty so we disable the Dropdown and
+		// prevent Dropdown to open if there are no children.
+		const openDropdown = hasChildren && !disabled && open;
 
 		return (
-			<div {...rest}>
-				<div {...wrapperProps}>
-					<DropDownButton
-						role="button"
-						className={buttonClassName}
-						css={css}
-						disabled={hasChildren ? disabled : true}
-						onClick={opened ? onClose : onOpen}
-						{...dropDownButtonProps}
-						{...ariaProps}
-					>
-						{iconComponent}
-						{title}
-					</DropDownButton>
-					<Transition
-						className={transitionContainerClassName}
-						visible={opened}
-						direction={oppositeDirection[adjustedDirection]}
-					>
-						<ContainerDiv className={dropdownListClassName} spotlightDisabled={!open} spotlightRestrict="self-only">
-							<Scroller skinVariants={skinVariants} className={css.scroller} onClick={onScrollerClick}>
-								<Group
-									role={null}
-									className={css.group}
-									onSelect={onSelect}
-									selected={selected}
-									{...groupProps}
-								>
-									{children || []}
-								</Group>
-							</Scroller>
-						</ContainerDiv>
-					</Transition>
-				</div>
+			<div {...calcAriaProps} {...rest}>
+				<DropdownButton
+					aria-label={ariaLabel}
+					className={buttonClassName}
+					css={css}
+					direction={direction}
+					disabled={hasChildren ? disabled : true}
+					dropdownOpened={openDropdown}
+					onClick={onOpen}
+					onClose={onClose}
+					open={openDropdown}
```

Comment:
I'm seeing `open` and `dropdownOpened`. It's a bit confusing.
If it's for deciding `icon` string, then I'd rather pass `icon` prop instead of this.